src/app/admin/products/ProductTable.tsx

```diff
@@ -53,46 +55,85 @@ function ProductTable({ products }: { products: Product[] }) {
             </div>
           )
         }
-        {sortedProducts.map((product) => {
-          return (
-            <Link
-              tabIndex={-1}
-              href={`/admin/products/${product.barcode}`}
-              key={product.barcode}
-            >
-              <div className="flex cursor-pointer justify-between border-b border-gray-200 px-4 py-3 transition-all hover:bg-stone-100">
-                <div className="flex min-h-full w-1/3 flex-col justify-between whitespace-nowrap">
-                  <h3 className="text-lg font-semibold">{product.name}</h3>
-                  <p className="text-sm text-stone-500">{product.barcode}</p>
-                </div>
-                {/*<div className="hidden w-1/3 flex-col items-end truncate xl:flex">
+        {sortedProducts.map((product) =>
+          compactMode ? (
+            <CompactProduct product={product} key={product.barcode} />
```

Comment:
Should this just be `<ProductRow product={product} key={product.barcode} compact={compactMode} />`?

Reply:
Assuming CompactProduct and NormalProduct are merged into one component. ProductRow could even be just a wrapper that does this logic.